### PR TITLE
A stab class for the ha-checker agent

### PR DESCRIPTION
**Pull request by aleksei-burlakov**
This class is supposed to implement ha-checker agents in the future.

### CODE REVIEW

**Comment by aleksei-burlakov** [comment]
@MalloZup could you please give your feedback?

File: cmd/agent/agent.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("You have registered your agent with arguments: " + strings.Join(args, " "))
+	},
+}
+
+var cmdStart = &cobra.Command{
+	Use:   "start [with arguments]",
+	Short: "Start the agent",
+	Long: `Start the
+agent`,
```
Comment (MalloZup) [inline cmd/agent/agent.go:32]: new line ?

File: cmd/agent/agent.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	cobra.CheckErr(cmdRegister.Execute())
+	cobra.CheckErr(cmdStart.Execute())
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+	rootCmd.AddCommand(cmdRegister)
+	rootCmd.AddCommand(cmdStart)
+}
+
```
Comment (MalloZup) [inline cmd/agent/agent.go:51]: this part we don't need to put it here.

we need to put and adapt it here: https://github.com/SUSE/console-for-sap-applications/blob/main/cmd/root.go#L34

File: cmd/agent/agent.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	Long: `Register the agent in the
+system`,
+	Args: cobra.MinimumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("You have registered your agent with arguments: " + strings.Join(args, " "))
+	},
+}
+
+var cmdStart = &cobra.Command{
+	Use:   "start [with arguments]",
```
Comment (MalloZup) [inline cmd/agent/agent.go:29]: we can remove `[with arguments]`

File: cmd/agent/agent.go (hunk 2 above)
Comment (MalloZup) [reply cmd/agent/agent.go:51]: the init part is placed on `root.go` file. in this we just need to implement the cobra part

File: cmd/agent/agent.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	Short: "Start the agent",
+	Long: `Start the
+agent`,
+	Args: cobra.MinimumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("You have started the agent with the arguments: " + strings.Join(args, " "))
```
Comment (MalloZup) [inline cmd/agent/agent.go:35]: we don't need to use different logic from:
https://github.com/SUSE/console-for-sap-applications/blob/main/cmd/webapp/webapp.go#L17-L35

take look there.

**Review by MalloZup — COMMENTED** [review]
good stab Aleksey :+1: 

I left some comment , but you got the basic layout good work!

File: cmd/agent/agent.go (hunk 1 above)
Comment (aleksei-burlakov) [reply cmd/agent/agent.go:32]: It's a stab class. I emphasize that a long comment might be multi-line.

Comment (aleksei-burlakov) [reply cmd/agent/agent.go:32]: I made it a single line.

File: cmd/agent/agent.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	"github.com/mitchellh/go-homedir"
+	"github.com/spf13/viper"
+
+	"github.com/SUSE/console-for-sap-applications/cmd/webapp"
+)
+
+var cfgFile string
+
+var cmdRegister = &cobra.Command{
+	Use:   "register [some arguments]",
```
[the remark this answers is not in the document]
Comment (aleksei-burlakov) [reply cmd/agent/agent.go:18]: Done.

File: cmd/agent/agent.go (hunk 3 above)
Comment (aleksei-burlakov) [reply cmd/agent/agent.go:29]: Done.

File: cmd/agent/agent.go (hunk 2 above)
Comment (aleksei-burlakov) [reply cmd/agent/agent.go:51]: I've removed the init function completely.

File: cmd/agent/agent.go (hunk 4 above)
Comment (aleksei-burlakov) [reply cmd/agent/agent.go]: Now I made it exactly the same.

File: cmd/root.go [hunk 6]
```diff
@@ -32,6 +33,8 @@ func init() {
 	cobra.OnInitialize(initConfig)
 	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.console-for-sap.yaml)")
 	rootCmd.AddCommand(webapp.NewWebappCmd())
+	rootCmd.AddCommand(agent.NewRegisterCmd())
+	rootCmd.AddCommand(agent.NewStartCmd())
```
Comment (stefanotorresi) [inline cmd/root.go:37]: This doesn't look quite right. The `start` and `register` commands should be children sub-commands of an intermediate `agent` command, which will be child of the root one.

**Review by stefanotorresi — CHANGES_REQUESTED** [review]
we need to rebase this against `main`, which now includes #8, and also a run of `make fmt`, which will fix some formatting.

File: cmd/agent/agent.go [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/spf13/viper"
+
```
Comment (stefanotorresi) [inline cmd/agent/agent.go:11]: unused packages will make the go compiler fail! ;)

File: cmd/root.go (hunk 6 above)
Comment (aleksei-burlakov) [reply cmd/root.go]: @stefanotorresi please, look again.

File: cmd/agent/agent.go (hunk 7 above)
Comment (aleksei-burlakov) [reply cmd/agent/agent.go]: Thanks, fixed!

**Review by stefanotorresi — APPROVED** [review]
Now it looks good!

**Pull request merged**